### DISCUSSION TITLE
How do the simultaneous sessions work?

### DISCUSSION THREAD

**Discussion by AwesomeDude091** [neutrinolabs/xrdp#2117] [Q&A]
Hello, I am a young computer scientist who is trying to understand multi sessions. After browsing through the code on this repository I could not find files were this behavior gets defined. Could somebody please point me towards those files and if they have spare time  explain how the concept works? Thanks in Advance, AwesomeDude091

**Comment by matt335672** [comment]
Hi @AwesomeDude091 

I'm not sure I understand your question exactly. What do you mean by "multi sessions"? It's not a term I'm familiar with.

**Reply by AwesomeDude091** [reply]
I mean the ability for multiple users to log in simultaneously. How is this implemented into Linux?

**Reply by matt335672** [reply]
I still don't follow you I'm afraid.

Linux has always supported multiple users. This isn't specifically xrdp functionality - it's baked into the operating system and inherited from Linux's UN*X roots. xrdp just sits on top of this.

Have you got a specific question related to xrdp?

**Reply by AwesomeDude091** [reply]
I guess I don't have a question to xrdp then, however what is the name of this baked feature? Also Thank you for the help :)

**Reply by matt335672** [reply]
Reading a bit between the lines, you need to mug up on operating system fundamentals.

[Here's a search](https://duckduckgo.com/?q=csca0201+"operating+systems") which has several useful links.
